packages/components/src/confirm-dialog/test/index.js

```diff
@@ -137,24 +148,27 @@ describe( 'Confirm', () => {
 			} );
 
 			it( 'should not render if dialog is closed by clicking the overlay, and the `onCancel` callback should be called', async () => {
+				const user = userEvent.setup();
 				const onCancel = jest.fn().mockName( 'onCancel()' );
 
 				render(
-					<ConfirmDialog onConfirm={ noop } onCancel={ onCancel }>
-						Are you sure?
-					</ConfirmDialog>
+					<div aria-label="Under the overlay">
+						<ConfirmDialog onConfirm={ noop } onCancel={ onCancel }>
+							Are you sure?
+						</ConfirmDialog>
+					</div>
 				);
 
 				const confirmDialog = screen.getByRole( 'dialog' );
 
-				//The overlay click is handled by detecting an onBlur from the modal frame.
-				// TODO: replace with `@testing-library/user-event`
-				fireEvent.blur( confirmDialog );
+				await user.click(
+					screen.getByLabelText( 'Under the overlay' )
+				);
 
 				await waitForElementToBeRemoved( confirmDialog );
 
 				expect( confirmDialog ).not.toBeInTheDocument();
-				expect( onCancel ).toHaveBeenCalled();
+				await waitFor( () => expect( onCancel ).toHaveBeenCalled() );
```

Comment:
Due to the extra logic and changes in timeout, a few more callbacks related to `onFocusOutside` will fire with a bigger delay